src/toil/batchSystems/slurm.py

```diff
@@ -647,107 +647,194 @@ def prepareSbatch(
             # Also any extra arguments from --slurmArgs or TOIL_SLURM_ARGS
             nativeConfig: str = self.boss.config.slurm_args  # type: ignore[attr-defined]
 
+            # For parsing user-provided option overrides (or self-generated
+            # options) we need a way to recognize long, long-with-equals, and
+            # short forms.
+            def option_detector(long: str, short: str | None = None) -> Callable[[str], bool]:
+                """
+                Get a function that returns true if it sees the long or short
+                option.
+                """
+                def is_match(option: str) -> bool:
+                    return option == f"--{long}" or option.startswith(f"--{long}=") or (short is not None and option == f"-{short}")
+                return is_match
+
+            def any_option_detector(options: list[str | tuple[str, str]]) -> Callable[[str], bool]:
```

Comment:
Suggestion: I think it might be worth having a small unit test just for this option detector.